scripts/auto-improve.sh

```
@@ -0,0 +1,83 @@
+#!/bin/bash
+set -euo pipefail
+
+# Configuration
+WAIT_DURATION=${CODERABBIT_WAIT_DURATION:-30}
+MAX_WAIT_TIME=${CODERABBIT_MAX_WAIT:-300}
+POLL_INTERVAL=${CODERABBIT_POLL_INTERVAL:-10}
+
+# Function to check if there are new commits from CodeRabbit
+check_for_new_commits() {
+    local initial_commit=$(git rev-parse HEAD)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)